app/decorators/user_decorator.rb

```diff
@@ -65,15 +49,18 @@ def qrcode(otp_secret_key)
     qrcode.as_png(size: 166).to_data_url
   end
 
+  def mobile_confirm(mobile)
+    SmsSenderNumberChangeJob.perform_later(user.mobile) if user.mobile.present?
+    user.mobile = mobile
+    user.mobile_confirmed_at = Time.zone.now
+    user.save
+  end
+
```

Comment:
This method does not belong in the decorator. The decorator is meant for methods that ask for a user's state or attribute that isn't mapped to a database column. A decorator method should not perform any actions that alter or affect the user.